test/cql-pytest/run.py

```diff
@@ -81,7 +81,7 @@ def run_with_temporary_dir(run_cmd_generator):
 def abort_run_with_temporary_dir(pid):
     tmpdir = pid_to_dir(pid)
     try:
-        os.kill(pid, 9)
+        os.killpg(pid, 9)
```

Comment:
unrelated: it's bad practice to use signal numbers since they can vary across architectures. Better use symbolic names.

Reply:
You're right that it's bad practice but for a different reasons - readers might not remember what this "9" is.
About changing numbers, that can't realistically happen. These numbers have not changed in 50 years ;-)

Reply:
They vary across architectures (not 9 though):

```

       Signal        x86/ARM     Alpha/   MIPS   PARISC   Notes
                   most others   SPARC
       ─────────────────────────────────────────────────────────────────
       SIGHUP           1           1       1       1
       SIGINT           2           2       2       2
       SIGQUIT          3           3       3       3
       SIGILL           4           4       4       4
       SIGTRAP          5           5       5       5
       SIGABRT          6           6       6       6
       SIGIOT           6           6       6       6
       SIGBUS           7          10      10      10
       SIGEMT           -           7       7      -
       SIGFPE           8           8       8       8
       SIGKILL          9           9       9       9
       SIGUSR1         10          30      16      16
       SIGSEGV         11          11      11      11
       SIGUSR2         12          31      17      17
       SIGPIPE         13          13      13      13
       SIGALRM         14          14      14      14
       SIGTERM         15          15      15      15
       SIGSTKFLT       16          -       -        7
       SIGCHLD         17          20      18      18
       SIGCLD           -          -       18      -
       SIGCONT         18          19      25      26
       SIGSTOP         19          17      23      24
       SIGTSTP         20          18      24      25
       SIGTTIN         21          21      26      27
       SIGTTOU         22          22      27      28
       SIGURG          23          16      21      29
       SIGXCPU         24          24      30      12
       SIGXFSZ         25          25      31      30
       SIGVTALRM       26          26      28      20
       SIGPROF         27          27      29      21
       SIGWINCH        28          28      20      23
       SIGIO           29          23      22      22
       SIGPOLL                                            Same as SIGIO
       SIGPWR          30         29/-     19      19
       SIGINFO          -         29/-     -       -
       SIGLOST          -         -/29     -       -
       SIGSYS          31          12      12      31
       SIGUNUSED       31          -       -       31
```

Reply:
The numbers 1, 2, 3, 9, and 15, have never changed on any Unix-like system for the last 50 years.
But you are right that using the name is nicer.

Reply:
Turns out the numbers 0, 1, 2, 3, 6, 9, 14 and 15 were standardized and officially can't change: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/kill.html
I'm surprised 11 wasn't (but people don't normally kill() with that signal anyway).

Reply:
I thought 11 was standardized by Spinal Tap.